src/prisma-generator.ts

```diff
@@ -1,123 +1,114 @@
 import { getDMMF, parseEnvValue } from '@prisma/internals';
 import { promises as fs } from 'fs';
 import Transformer from './transformer';
 import removeDir from './utils/removeDir';
+import {
+  addMissingInputObjectTypes,
+  handleMongoDbRawOperationsAndQueries,
+} from './helpers';
 
 export async function generate(options: GeneratorOptions) {
-  const outputDir = parseEnvValue(options.generator.output as EnvValue);
-  await fs.mkdir(outputDir, { recursive: true });
-  await removeDir(outputDir, true);
+  await handleGeneratorOutputValue(options.generator.output as EnvValue);
 
-  const prismaClientProvider = options.otherGenerators.find(
-    (it) => parseEnvValue(it.provider) === 'prisma-client-js',
+  const prismaClientGeneratorConfig = getGeneratorConfigByProvider(
+    options.otherGenerators,
+    'prisma-client-js',
   );
 
   const prismaClientDmmf = await getDMMF({
     datamodel: options.datamodel,
-    previewFeatures: prismaClientProvider?.previewFeatures,
+    previewFeatures: prismaClientGeneratorConfig?.previewFeatures,
   });
 
-  const dataSource = options.datasources?.[0];
+  checkForCustomPrismaClientOutputPath(prismaClientGeneratorConfig);
+
+  await generateEnumSchemas(
+    prismaClientDmmf.schema.enumTypes.prisma,
+    prismaClientDmmf.schema.enumTypes.model ?? [],
+  );
+
+  const models = prismaClientDmmf.datamodel.models;
+  const modelOperations = prismaClientDmmf.mappings.modelOperations;
+  const inputObjectTypes = prismaClientDmmf.schema.inputObjectTypes.prisma;
+  const outputObjectTypes = prismaClientDmmf.schema.outputObjectTypes.prisma;
 
-  Transformer.isDefaultPrismaClientOutput =
-    prismaClientProvider?.isCustomOutput ?? false;
+  const dataSource = options.datasources?.[0];
+  Transformer.provider = dataSource.provider;
 
-  if (prismaClientProvider?.isCustomOutput) {
-    Transformer.prismaClientOutputPath =
-      prismaClientProvider?.output?.value ?? '';
+  // TODO: remove once Prisma fix this issue: https://github.com/prisma/prisma/issues/14900
+  if (dataSource.provider === 'mongodb') {
+    handleMongoDbRawOperationsAndQueries(
+      modelOperations,
+      outputObjectTypes,
+      inputObjectTypes,
+    );
   }
 
-  Transformer.setOutputPath(outputDir);
+  addMissingInputObjectTypes(inputObjectTypes, outputObjectTypes, models);
+  await generateObjectSchemas(inputObjectTypes);
 
-  const enumTypes = [
-    ...prismaClientDmmf.schema.enumTypes.prisma,
-    ...(prismaClientDmmf.schema.enumTypes.model ?? []),
-  ];
-  const enumNames = enumTypes.map((enumItem) => enumItem.name);
-  Transformer.enumNames = enumNames ?? [];
-  const enumsObj = new Transformer({
-    enumTypes,
-  });
-  await enumsObj.printEnumSchemas();
+  await generateModelSchemas(modelOperations);
+}
 
-  const inputObjectTypes = prismaClientDmmf.schema.inputObjectTypes.prisma;
-  const rawOpsMap: { [name: string]: string } = {};
+async function handleGeneratorOutputValue(generatorOuputValue: EnvValue) {
```

Comment:
typo in `generatorOuputValue`
should be `generatorOutputValue`

missing `t`

Reply:
Great catch! Fixed ✅